### DISCUSSION TITLE
Dataset for the User's Access activates

### DISCUSSION THREAD

**Discussion by omaroudat** [riclolsen/json-scada#51] [Q&A] [answered]
Hi Mr. Ricardo, 

I installed the system in My Windows 10 and it is up and running. 

I created some users and roles and I am looking to generate a dataset for the user's activities when they trying to access the resources. 

I saw that the one of the features is  User auth/Role-based Access Control (RBAC)

Any luck of generate a dataset regarding the user's roles and the resources to which they have access ?

**Comment by riclolsen** [comment, marked as answer]
Only admin users can see the userActions log in the UI.

You can use the included MongoDB Compass to export datasets, connect to

    mongodb://localhost:27017/json_scada?replicaSet=rs1&readPreference=primary&appname=MongoDB%20Compass&ssl=false

Enter json_scada database. Query the userActions collection, you can filter by username, date, etc. Export data to CSV.

**Reply by omaroudat** [reply]
Thank you for your response Mr. Ricardo. 

This is the table that i need, but I need a huge rows in this table so i can work on these data using the machine learning. 
Do i have any luck automate this process to have a lot of these data in the UserAction table ?

**Reply by riclolsen** [reply]
This table is the result of real user actions. When used in production this type of information is confidential.
So, it is difficult to obtain real data without producing it in your own enterprise environment. 
I do not have a tool to create this data Anyway, automated data may not be useful/meaningful for ML processing.

**Reply by omaroudat** [reply]
Thank you dear

One more question please, can I have another columns in this table showing the Network traffic details, IP Address, Port, etc ?

**Reply by riclolsen** [reply]
No, this type of information is planned to appear in the protocolConnections table. Some protocols already write some statistics there but it is not available in the UI yet.
